generate-api-types.js

```diff
@@ -231,11 +233,11 @@ typeFile += '\n' + enumDefs + '\n';
 
 fs.writeFileSync(openAPIGeneratedPath, typeFile);
 const reExporterFile = generateReExporterFile(typeFile, args['types-dir'], enumLookup);
-const schemasPath = path.join(args['project-root'], args['types-dir'], 'schemas.ts');
+const schemasPath = path.join(args['project-root'], args['types-dir'], 'schemas.d.ts');
```

Comment:
Without the `.d` is correct. The TS file uses exports, making it not be a definition file. It would also not allow enum reexporting.

Reply:
I get this when running it locally with the `.d`
```
(nook-app-py3.11)  app/ sync-retries $ generate-api-types.js --project-root ./frontend/src --oas-url http://localhost:8000/openapi.json --oas-command './manage.py openapi' --command-cwd ./backend
(node:91909) ExperimentalWarning: Import assertions are not a stable feature of the JavaScript language. Avoid relying on their current behavior and syntax as those might change in a future version of Node.js.
(Use `node --trace-warnings ...` to show where the warning was created)
(node:91909) ExperimentalWarning: Importing JSON modules is an experimental feature and might change at any time
node:fs:603
  handleErrorFromBinding(ctx);
  ^

Error: ENOENT: no such file or directory, open 'frontend/src/types/openapi.ts'
    at Object.openSync (node:fs:603:3)
    at Object.readFileSync (node:fs:471:35)
    at getSchemaHash (file:///Users/luis/openapi-type-generator/generate-api-types.js:167:26)
    at file:///Users/luis/openapi-type-generator/generate-api-types.js:215:24
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: 'frontend/src/types/openapi.ts'
}

Node.js v18.18.0
```

perhaps I'm doing something wrong then?

Reply:
I guess it's not robust against there not being an existing openapi.ts file?